### PR TITLE
Add onClick option

### PR DESCRIPTION
**Pull request by SanderMuller**
**New feature - onClick**
Add an on click listener similar to the hover option (but flatter).
```$bar->options()->setOnClick(new \Zend\Json\Expr('graphClickEvent'));```

**Other changes:**
- Updated dev from master
- Created getters to pass unit tests
- All tests now pass

### CODE REVIEW

**Comment by halfpastfouram** [comment]
Thanks for your PR! Looks great. I'd like to split the StringUtils delegate. Could you move the new methods that have nothing to do with strings into a new delegate names `NumberUtils`?

**Comment by SanderMuller** [comment]
I have updated the PR with your request.
Please be advised that my local unit tests fail on Test\Options\Elements\LineTest::testJsonSerialize (outside the scope of this PR)

File: composer.json [hunk 1]
```diff
@@ -8,8 +8,8 @@
   ],
   "homepage": "http://github.com/halfpastfouram/PHPChartJS",
   "type": "package",
-  "version": "v1.1.0",
-  "license": "AGPL-3.0-or-later",
+  "version": "dev-master",
+  "license": "AGPL 3.0",
```
Comment (halfpastfouram) [inline composer.json]:
```suggestion
  "license": "AGPL-3.0-or-later",
```

File: src/Options/Legend.php [hunk 2]
```diff
@@ -59,6 +59,14 @@ public function isDisplay()
         return $this->display;
     }
 
+    /**
+     * @return bool
+     */
+    public function getDisplay()
```
Comment (halfpastfouram) [inline src/Options/Legend.php]:
```suggestion
    public function isDisplay()
```

File: src/Options/Legend.php [hunk 3]
```diff
@@ -99,6 +107,14 @@ public function isFullWidth()
         return $this->fullWidth;
     }
 
+    /**
+     * @return bool
+     */
+    public function getFullWidth()
```
Comment (halfpastfouram) [inline src/Options/Legend.php]:
```suggestion
    public function isFullWidth()
```

File: src/Options/Legend.php [hunk 4]
```diff
@@ -171,6 +187,14 @@ public function isReverse()
         return $this->reverse;
     }
 
+    /**
+     * @return bool
+     */
+    public function getReverse()
```
Comment (halfpastfouram) [inline src/Options/Legend.php]:
```suggestion
    public function isReverse()
```

File: src/Options/Legend/Labels.php [hunk 5]
```diff
@@ -203,6 +203,14 @@ public function isUsePointStyle()
         return $this->usePointStyle;
     }
 
+    /**
+     * @return bool
+     */
+    public function getUsePointStyle()
```
Comment (halfpastfouram) [inline src/Options/Legend/Labels.php]:
```suggestion
    public function isUsePointStyle()
```

File: src/Options/Scale.php [hunk 6]
```diff
@@ -160,6 +160,14 @@ public function isDisplay()
         return $this->display;
     }
 
+    /**
+     * @return bool
+     */
+    public function getDisplay()
```
Comment (halfpastfouram) [inline src/Options/Scale.php]:
```suggestion
    public function isDisplay()
```

File: src/Options/Scale.php [hunk 7]
```diff
@@ -200,6 +208,14 @@ public function isStacked()
         return $this->stacked;
     }
 
+    /**
+     * @return bool
+     */
+    public function getStacked()
```
Comment (halfpastfouram) [inline src/Options/Scale.php]:
```suggestion
    public function isStacked()
```

File: src/Options/Scales/GridLines.php [hunk 8]
```diff
@@ -85,6 +87,14 @@ public function isDisplay()
         return $this->display;
     }
 
+    /**
+     * @return bool
+     */
+    public function getDisplay()
```
Comment (halfpastfouram) [inline src/Options/Scales/GridLines.php]:
```suggestion
    public function isDisplay()
```

File: src/Options/Scales/GridLines.php [hunk 9]
```diff
@@ -195,6 +205,14 @@ public function isDrawBorder()
         return $this->drawBorder;
     }
 
+    /**
+     * @return bool
+     */
+    public function getDrawBorder()
```
Comment (halfpastfouram) [inline src/Options/Scales/GridLines.php]:
```suggestion
    public function isDrawBorder()
```

File: src/Options/Scales/GridLines.php [hunk 10]
```diff
@@ -215,6 +233,14 @@ public function isDrawOnChartArea()
         return $this->drawOnChartArea;
     }
 
+    /**
+     * @return bool
+     */
+    public function getDrawOnChartArea()
```
Comment (halfpastfouram) [inline src/Options/Scales/GridLines.php]:
```suggestion
    public function isDrawOnChartArea()
```

File: src/Options/Scales/GridLines.php [hunk 11]
```diff
@@ -227,6 +253,14 @@ public function setDrawOnChartArea($drawOnChartArea)
         return $this;
     }
 
+    /**
+     * @return bool
+     */
+    public function getDrawTicks()
```
Comment (halfpastfouram) [inline src/Options/Scales/GridLines.php]:
```suggestion
    public function isDrawTicks()
```

File: src/Options/Scales/GridLines.php [hunk 12]
```diff
@@ -315,6 +349,14 @@ public function isOffsetGridLines()
         return $this->offsetGridLines;
     }
 
+    /**
+     * @return bool
+     */
+    public function getOffsetGridLines()
```
Comment (halfpastfouram) [inline src/Options/Scales/GridLines.php]:
```suggestion
    public function isOffsetGridLines()
```

File: src/Options/Scales/ScaleLabel.php [hunk 13]
```diff
@@ -52,6 +52,14 @@ public function isDisplay()
         return $this->display;
     }
 
+    /**
+     * @return bool
+     */
+    public function getDisplay()
```
Comment (halfpastfouram) [inline src/Options/Scales/ScaleLabel.php]:
```suggestion
    public function isDisplay()
```

File: src/Options/Scales/Ticks.php [hunk 14]
```diff
@@ -134,6 +134,14 @@ public function isBeginAtZero()
         return $this->beginAtZero;
     }
 
+    /**
+     * @return bool
+     */
+    public function getBeginAtZero()
```
Comment (halfpastfouram) [inline src/Options/Scales/Ticks.php]:
```suggestion
    public function isBeginAtZero()
```

File: src/Options/Scales/Ticks.php [hunk 15]
```diff
@@ -174,6 +182,14 @@ public function isAutoSkip()
         return $this->autoSkip;
     }
 
+    /**
+     * @return bool
+     */
+    public function getAutoSkip()
```
Comment (halfpastfouram) [inline src/Options/Scales/Ticks.php]:
```suggestion
    public function isAutoSkip()
```

File: src/Options/Scales/Ticks.php [hunk 16]
```diff
@@ -234,6 +250,14 @@ public function isDisplay()
         return $this->display;
     }
 
+    /**
+     * @return bool
+     */
+    public function getDisplay()
```
Comment (halfpastfouram) [inline src/Options/Scales/Ticks.php]:
```suggestion
    public function isDisplay()
```

File: src/Options/Scales/Ticks.php [hunk 17]
```diff
@@ -394,6 +418,14 @@ public function isMirror()
         return $this->mirror;
     }
 
+    /**
+     * @return bool
+     */
+    public function getMirror()
```
Comment (halfpastfouram) [inline src/Options/Scales/Ticks.php]:
```suggestion
    public function isMirror()
```

File: src/Options/Scales/Ticks.php [hunk 18]
```diff
@@ -434,6 +466,14 @@ public function isReverse()
         return $this->reverse;
     }
 
+    /**
+     * @return bool
+     */
+    public function getReverse()
```
Comment (halfpastfouram) [inline src/Options/Scales/Ticks.php]:
```suggestion
    public function isReverse()
```

File: src/Options/Title.php [hunk 19]
```diff
@@ -67,6 +67,14 @@ public function isDisplay()
         return $this->display;
     }
 
+    /**
+     * @return bool
+     */
+    public function getDisplay()
```
Comment (halfpastfouram) [inline src/Options/Title.php]:
```suggestion
    public function isDisplay()
```

File: src/Options/Title.php [hunk 20]
```diff
@@ -107,6 +115,14 @@ public function isFullWidth()
         return $this->fullWidth;
     }
 
+    /**
+     * @return bool
+     */
+    public function getFullWidth()
```
Comment (halfpastfouram) [inline src/Options/Title.php]:
```suggestion
    public function isFullWidth()
```

File: src/Options/Tooltips.php [hunk 21]
```diff
@@ -184,6 +184,14 @@ public function isEnabled()
         return $this->enabled;
     }
 
+    /**
+     * @return bool
+     */
+    public function getEnabled()
```
Comment (halfpastfouram) [inline src/Options/Tooltips.php]:
```suggestion
    public function isEnabled()
```

File: src/Options/Tooltips.php [hunk 22]
```diff
@@ -244,6 +252,14 @@ public function isIntersect()
         return $this->intersect;
     }
 
+    /**
+     * @return bool
+     */
+    public function getIntersect()
```
Comment (halfpastfouram) [inline src/Options/Tooltips.php]:
```suggestion
    public function isIntersect()
```

File: src/Options/Tooltips.php [hunk 23]
```diff
@@ -784,6 +800,14 @@ public function isDisplayColors()
         return $this->displayColors;
     }
 
+    /**
+     * @return bool
+     */
+    public function getDisplayColors()
```
Comment (halfpastfouram) [inline src/Options/Tooltips.php]:
```suggestion
    public function isDisplayColors()
```

**Pull request merged**